kernel_gateway/services/sessions/handlers.py

```diff
@@ -0,0 +1,11 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import notebook.services.sessions.handlers as notebook_handlers
+from ...mixins import TokenAuthorizationMixin, CORSMixin
+
+default_handlers = []
+for path, cls in notebook_handlers.default_handlers:
+    # Everything should have CORS and token auth
+    bases = (TokenAuthorizationMixin, CORSMixin, cls)
```

Comment:
It's good that you're protecting the endpoints like we did in other cases. I wonder if the "list" action `GET /api/sessions` needs to become an option like `GET /api/kernels` did. That way, a public server could be stood up without different clients being able to "discover" one another simply by listing the sessions.